src/rsp_restspawner/util.py

```diff
@@ -0,0 +1,67 @@
+"""General utility functions."""
+
+import os
+from typing import Any, Dict
+
+import yaml
+
+_config: Dict[str, Any] = {}
+
+
+def to_camel_case(string: str) -> str:
+    """Convert a string to camel case.
+
+    Originally written for use with Pydantic as an alias generator so that the
+    model can be initialized from camel-case input (such as Kubernetes
+    objects).
+
+    Parameters
+    ----------
+    string
+        Input string
+
+    Returns
+    -------
+    str
+        String converted to camel-case with the first character in lowercase.
+    """
+    components = string.split("_")
+    return components[0] + "".join(c.title() for c in components[1:])
+
+
+def get_namespace() -> str:
+    ns_path = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+    if os.path.exists(ns_path):
+        with open(ns_path) as f:
+            ns = f.read().strip()
+    else:
+        ns = "userlabs"
+    return ns
```

Comment:
True, *if* we get namespace, or at least internal URL, data from the user status query.

Reply:
Which we will.

Reply:
Done.

Reply:
Actually, we do need this function:

```
             import os
              import rsp_restspawner  # Baked into image
              from rsp_restspawner.util import get_namespace

              # Turn off concurrent spawn limit
              c.JupyterHub.concurrent_spawn_limit = 0
              #
              # Set internal url
              c.JupyterHub.hub_connect_url = (
                  "http://hub." +
                  f"{get_namespace()}:" +
                  f"{os.environ['HUB_SERVICE_PORT']}"
              )
  ```

Reply:
(I'll see what happens if we leave it out, but I think we won't be able to talk from the lab to the hub)

Reply:
Well, I guess I can see what happens if we set it to the external address...

```
[E 2023-03-09 18:59:41.894 SingleUserLabApp mixins:484] Failed to connect to my Hub at http://hub:8081/nb/hub/api (attempt 2/5). Is it running?
```

Reply:
Aha.  We're actually poking the configuration into the container, mounted as a file.  So instead of doing anything to detect the namespace, I can just load the config object from the yaml (which we already have a convenience function to do) and then just read it out of the config object.